src/App.jsx

```diff
@@ -13,42 +13,64 @@ function App() {
   const [isLoading, setIsLoading] = useState(true)
   const [error, setError] = useState('')
 
-  useEffect(() => {
-    const myPromise = new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve( JSON.parse(localStorage.getItem('savedTodoList')) || [] )
-      }, 2000)
-    })
-    myPromise.then((result) => {
-      setTodoList(result)
+  const fetchData = async () => {
+    const options = {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`
+      },
+      url: `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`,
+    }
+
+    try {
+      const response = await fetch(options.url, options)
+      if ( !response.ok ) {
+        throw new Error(`Error: ${response.status}`)
+      }
+      const data = await response.json()
+
+      const todos = data.records.map((item) => {
+        return {
+          id: item.id,
+          title: item.fields.title,
+          completedAt: item.fields.completedAt,
+        }
+      })
+      setTodoList(todos)
       setIsLoading(false)
-    })
-    
-    myPromise.catch((error) => {
+
+    } catch (error) {
       setIsLoading(false)
-      setError("Something went wrong")
+      setError("Something went wrong with the API")
       console.error(error)
-    })
+    }
+  }
+
+
+  useEffect(() => {
+    (async ()=>{
```

Comment:
nit: no need for all the async/await, you can just call `fetchData()`

This is only in this case, where the only function being called is `fetchData`